client/my-sites/email/email-management/home/utils.ts

```diff
@@ -219,12 +220,21 @@ export function recordInboxNewMailboxUpsellClickEvent() {
 }
 
 /**
- * Tracks an event for the key 'calypso_inbox_upsell'.
+ * Tracks an event for the key 'calypso_{source}_upsell', where {source} defaults to "email".
+ * For upsell triggered from the inbox, the event 'calypso_inbox_upsell' will be tracked.
  *
+ * @param source - source generating the event.
  * @param context context, where this event was logged.
  */
-export function recordInboxUpsellTracksEvent( context: string | null = null ) {
-	recordTracksEvent( 'calypso_inbox_upsell', {
+export function recordEmailUpsellTracksEvent(
+	source: string | null = null,
+	context: string | null = null
+) {
+	if ( ! source ) {
+		source = 'email';
+	}
```

Comment:
What about:
```suggestion
    source = source || 'email';
```

But after analyzing more the method parameters, I am not sure why those parameters are that complicated, why not just set the method parameters as:
`export function recordEmailUpsellTracksEvent( source: string, context: string )`

Maybe I'm confused but seems to me that this function is never called with null parameters.

Reply:
`source` will be `null` when user clicks the CTA on the **Upgrades > Email** page (rather than **Inbox**).  But you're right, this can be simplified! I have updated the code to supply `email` as the default value in the function signature itself :)